### DISCUSSION TITLE
Progressive Delivery

### DISCUSSION THREAD

**Discussion by chris-short** [open-gitops/project#93] [General]
I'd like to start a discussion around this topic to get a better idea of what people are seeing/doing around progressive delivery (or minimizing blast radius of improper configurations).

There are a number of progressive delivery tools out there. Some work really well with existing GitOps controllers. I think it would make sense for us to pull together a best practices doc or guide around this subject. We can highlight tools but, we shouldn't say one tool is the de facto tool because we shouldn't be THAT opinionated.

I'd love to hear more around what folks are doing in this space and would like to work with other contributors on establishing some documentation on this topic.

**Comment by gregswift** [comment]
[Relevant article from AWS on how they do Release Waves](https://aws.amazon.com/builders-library/automating-safe-hands-off-deployments/)

**Reply by chris-short** [reply]
AWS architecture is massive and from what I know this article is still accurate and could be a guiding example for large deployments.

**Comment by chris-short** [comment]
https://www.youtube.com/watch?v=OIWUyKuuytQ

**Comment by chris-short** [comment]
I've seen folks doing multiple branches that split up the prod environment to a manageable blue/green. Some the branch name would be (as an example): product v.X.Y.Z-wave-1, 2, 3, and so on. That's certainly a valid GitOps pattern if it's automated, in my opinion.
